src/h5web/visualizations/shared/AxisSystem.tsx

```diff
@@ -61,6 +58,14 @@ function AxisSystem(props: Props): JSX.Element {
     ),
   };
 
+  const xAxisScale = axisToCanvas.x.copy();
```

Comment:
To avoid copying the scale, you could add a hook called `useAbscissaScaleFn` and use it here and inside `useAbscissaScale`.